### PR TITLE
Bug 1735728: Add description to operator user facing CRDs.

### PR DESCRIPTION
**Pull request by pliurh**

### CODE REVIEW

File: deploy/crds/sriovnetwork_v1_sriovnetwork_crd.yaml [hunk 1]
```diff
@@ -26,22 +26,30 @@ spec:
         metadata:
           type: object
         spec:
+          description: Specification describing how a NetworkAttachmentDefinition
```
Comment (zshi-redhat) [inline deploy/crds/sriovnetwork_v1_sriovnetwork_crd.yaml]: Shall we also change the file under `manifests/4.2/manifests/4.2/sriov-network-operator-sriovnetwork.crd.yaml`?
before this PR, they have same content.

File: pkg/apis/sriovnetwork/v1/sriovnetworknodestate_types.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ type SriovNetworkNodeStateStatus struct {
 
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
+// +kubebuilder:subresource:status
```
[the remark this answers is not in the document]
Comment (pliurh) [reply pkg/apis/sriovnetwork/v1/sriovnetworknodestate_types.go]: There is a bug in operator-sdk, which makes errors the generated CRD. Therefore, we have to do manual change to the generated CRD.

File: pkg/apis/k8s/v1/zz_generated.deepcopy.go [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // +build !ignore_autogenerated
 
-// Code generated by deepcopy-gen. DO NOT EDIT.
+// Code generated by operator-sdk. DO NOT EDIT.
```
[the remark this answers is not in the document]
Comment (pliurh) [reply pkg/apis/k8s/v1/zz_generated.deepcopy.go]: No. It caused by the operator-sdk update.

File: deploy/crds/sriovnetwork_v1_sriovnetworknodepolicy_crd.yaml [hunk 4]
```diff
@@ -10,8 +10,6 @@ spec:
     plural: sriovnetworknodepolicies
     singular: sriovnetworknodepolicy
   scope: Namespaced
-  subresources:
```
[the remark this answers is not in the document]
Comment (pliurh) [reply deploy/crds/sriovnetwork_v1_sriovnetworknodepolicy_crd.yaml]: Correct

File: deploy/crds/sriovnetwork_v1_sriovnetwork_crd.yaml [hunk 5]
```diff
@@ -26,22 +26,30 @@ spec:
...
             trust:
+              description: VF trust mode
               type: boolean
             vlan:
+              description: VLAN ID to assign for the VF. Defaults to 0.
               format: int64
-              type: integer
-              minimum: 0
               maximum: 4096
+              minimum: 0
```
[the remark this answers is not in the document]
Comment (pliurh) [reply deploy/crds/sriovnetwork_v1_sriovnetwork_crd.yaml]: The CRD file is auto generated by operator-sdk. I don't think the order matters.

File: deploy/crds/sriovnetwork_v1_sriovnetwork_crd.yaml (hunk 1 above)
Comment (pliurh) [reply deploy/crds/sriovnetwork_v1_sriovnetwork_crd.yaml]: Yes. Will do.

**Pull request merged**